kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KopBrokerLookupManager.java

```diff
@@ -135,6 +141,12 @@ private String getAdvertisedListener(InetSocketAddress internalListenerAddress,
             log.error("No node for broker {} under loadBalance", internalListenerAddress);
             return null;
         }
+        if (serviceLookupData.get().getProtocol(KafkaProtocolHandler.PROTOCOL_NAME).isPresent()
+                && serviceLookupData.get().getProtocol(KafkaProtocolHandler.PROTOCOL_NAME).get()
+                        .equals(selfAdvertisedListeners)){
+            // the topic is owned by this broker, cache the look up result
+            LOOKUP_CACHE.put(topic, CompletableFuture.completedFuture(internalListenerAddress));
+        }
 
         return serviceLookupData.get().getProtocol(KafkaProtocolHandler.PROTOCOL_NAME).map(kafkaAdvertisedListeners ->
```

Comment:
I think we can simplify the code by

```java
        return serviceLookupData.get().getProtocol(KafkaProtocolHandler.PROTOCOL_NAME).map(kafkaAdvertisedListeners -> {
            if (kafkaAdvertisedListeners.equals(selfAdvertisedListeners)) {
                // the topic is owned by this broker, cache the look up result
                LOOKUP_CACHE.put(topic, CompletableFuture.completedFuture(internalListenerAddress));
            }
            return Optional.ofNullable(advertisedEndPoint)
                    .map(endPoint -> EndPoint.findListener(kafkaAdvertisedListeners, endPoint.getListenerName()))
                    .orElse(EndPoint.findFirstListener(kafkaAdvertisedListeners));
        }).orElseThrow(() -> new IllegalStateException(
                "No kafkaAdvertisedListeners found in broker " + internalListenerAddress));
```

There is no need to call `serviceLookupData.get().getProtocol(KafkaProtocolHandler.PROTOCOL_NAME)` for multiple times.

Reply:
Done